src/osd/PrimaryLogPG.cc

```diff
@@ -1589,25 +1591,72 @@ int PrimaryLogPG::do_scrub_ls(const MOSDOp *m, OSDOp *osd_op)
     dout(10) << " corrupted scrub_ls_arg_t" << dendl;
     return -EINVAL;
   }
+
   int r = 0;
   scrub_ls_result_t result = {.interval = info.history.same_interval_since};
+
   if (arg.interval != 0 && arg.interval != info.history.same_interval_since) {
     r = -EAGAIN;
-  } else if (!scrubber.store) {
-    r = -ENOENT;
-  } else if (arg.get_snapsets) {
-    result.vals = scrubber.store->get_snap_errors(get_pgid().pool(),
-						  arg.start_after,
-						  arg.max_return);
   } else {
-    result.vals = scrubber.store->get_object_errors(get_pgid().pool(),
-						    arg.start_after,
-						    arg.max_return);
+    bool store_queried = m_scrubber->get_store_errors(arg, result);
+    if (!store_queried) {
+      // the scrubber's store is not initialized
+      r = -ENOENT;
+    }
   }
-  encode(result, osd_op->outdata);
+  encode(result, osd_op->outdata);  // RRR really? even if no store?
```

Comment:
In fact, in Objecter::C_ObjectOperation_scrub_ls we don't appear to decode the payload if unsuccessful, so we presumably could skip this encode.  I really don't want to leave these questions in the code, btw.  If we don't know the answer now, readers in the future certainly won't be able to figure it out.